ocrd/ocrd/decorators.py

```diff
@@ -46,9 +56,21 @@ def ocrd_cli_wrap_processor(processorClass, ocrd_tool=None, mets=None, working_d
             raise Exception(msg)
         resolver = Resolver()
         workspace = resolver.workspace_from_url(mets, working_dir)
-        # TODO once we implement 'overwrite' CLI option and mechanism, disable the
-        # `output_file_grp_ check by setting to False-y value if 'overwrite' is set
-        report = WorkspaceValidator.check_file_grp(workspace, kwargs['input_file_grp'], kwargs['output_file_grp'])
+        page_id = kwargs.get('page_id')
+        if overwrite:
+            if 'output_file_grp' not in kwargs or not kwargs['output_file_grp']:
+                raise Exception("--overwrite requires --output-file-grp")
+            LOG.debug("Removing files because of --overwrite")
+            for grp in kwargs['output_file_grp'].split(','):
+                if page_id:
+                    for one_page_id in kwargs['page_id'].split(','):
+                        LOG.debug("Removing files in output file group %s with page ID %s", grp, one_page_id)
+                        for file in workspace.mets.find_files(pageId=one_page_id, fileGrp=grp):
+                            workspace.remove_file(file, force=True, keep_file=False, page_recursive=True)
+                else:
+                    LOG.debug("Removing all files in output file group %s ", grp)
+                    workspace.remove_file_group(grp, recursive=True, force=True, keep_files=False, page_recursive=True, page_same_group=True)
+        report = WorkspaceValidator.check_file_grp(workspace, kwargs['input_file_grp'], kwargs['output_file_grp'], page_id)
```

Comment:
And there's yet **another problem**: When processing itself fails (for some reason), now the deletion is effective immediately in the filesystem, but the new METS does _not_ get written. Thus, stale fptrs remain which have to be removed with an extra `workspace prune-files`.

```suggestion
            workspace.save_mets()
        report = WorkspaceValidator.check_file_grp(workspace, kwargs['input_file_grp'], kwargs['output_file_grp'], page_id)
```

Reply:
Good catch. Failing processors is something we should test more.